### PR TITLE
Fix vector tiles not loading when html is opened via file://

### PR DESCRIPTION
**Pull request by sbachinin**
Fixes https://github.com/maplibre/maplibre-gl-js/issues/3680

### CODE REVIEW

**Comment by HarelM** [comment]
Can you please add a changelog item?
I'm not sure this is worth a test, but the coverage report rightfully reports that this change is only partially covered, your call.

Please keep the template next time you open a PR, it has reminders of things that should be done as part of the PR, mainly for me to remember to review.

File: src/util/actor.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export class Actor implements IActor {
     receive(message: {data: MessageData}) {
         const data = message.data;
         const id = data.id;
-        if (data.origin !== location.origin) {
+        if (data.origin !== 'file://' && location.origin !== 'file://' && data.origin !== location.origin) {
```
Comment (HarelM) [inline src/util/actor.ts:132]: Seems like the test is not covering all options of this condition, also I think this condition can be simplified, can't it?
The check is that both are not file protocol and are not the same, I tend to think it's enough to check one of them for file and not the same, not sure though...
Maybe not, IDK...

Comment (sbachinin) [reply src/util/actor.ts:132]: I really don't know how this line can be partly covered. There is at least one test case where all parts of the condition are true, therefore the line is fully executed.

This solution works only with both `!== file://` checks

Comment (HarelM) [reply src/util/actor.ts:132]: Ok, thanks for taking the time to check this!

**Review by HarelM — APPROVED** [review]

**Pull request merged**